tabcmd/commands/datasources_and_workbooks/delete_command.py

```diff
@@ -20,33 +21,45 @@ def run_command(args):
         logger.debug("Launching command")
         session = Session()
         server = session.create_session(args)
-        if args.workbook:
-            # filter match the name and find id
+        located_workbook = None
+        located_datasource = None
+        req_option = TSC.RequestOptions()
+        possible_workbook = args.name if args.name else args.workbook
+        possible_datasource = args.name if args.name else args.datasource
+        try:
+            req_option.filter.add(TSC.Filter(TSC.RequestOptions.Field.Name,
```

Comment:
tiny optimization: you can build the request options here just once without knowing if it's a workbook or datasource, something like
possible_item = args.name or args.workbook or args.datasource
then use the same request option in both calls

Reply:
Thanks, Fixed it!